python/pyarrow/tests/test_pandas.py

```diff
@@ -3321,9 +3321,12 @@ def test_cast_timestamp_unit():
     assert result.equals(expected)
 
 
-def test_struct_with_timestamp_tz():
+def test_nested_with_timestamp_tz():
     # ARROW-7723
     ts = pd.Timestamp.now()
+    # This is used for verifying timezone conversion to micros are not
+    # important
+    ts_dt = ts.to_pydatetime().replace(microsecond=0)
```

Comment:
Why is this needed? (or, I suppose this is only needed if the unit is 's' or 'ms' ? If so, I would rather do this only in those cases, and for 'us' unit actually check the microseconds are also correct)

Reply:
> There is one failure (which involves datetimes with timezones, but can't directly understand if it also involves structs):
> 
> ```
> ======================================================================
> FAIL: test_grouped_over_window_with_key (pyspark.sql.tests.test_pandas_grouped_map.GroupedMapInPandasTests)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/spark/python/pyspark/sql/tests/test_pandas_grouped_map.py", line 588, in test_grouped_over_window_with_key
>     self.assertTrue(all([r[0] for r in result]))
> AssertionError: False is not true
> 
> ----------------------------------------------------------------------
> Ran 21 tests in 141.194s
> 
> FAILED (failures=1)
> ```
> 
> https://github.com/apache/spark/blob/7fda184f0fc39613fb68e912c189c54b93c638e6/python/pyspark/sql/tests/test_pandas_grouped_map.py#L546-L574

Yeah this seems strange.  @BryanCutler do you have a sense of what is going on here?

Reply:
👋  another observation from the R side which may or may not be relevant here: https://github.com/sparklyr/sparklyr/issues/2439

Reply:
I'm not too sure about the failure and the message isn't helpful at all. The test does have a struct with 2 timestamps in it. I should be able to at least get a better error message right now, and then I can look into what's going on.

Reply:
sorry I realize I responded to spark test failure on wrong chain.  @jorisvandenbossche I addressed your original feedback here.